### PR TITLE
feat(core): adds prefetch and prefetchKind to Link

### PR DESCRIPTION
**Pull request by deini**
## What/Why?
Adds the following props to our  custom `Link` component

```ts
interface PrefetchOptions {
  prefetch?: 'hover' | 'viewport' | 'none';
  prefetchKind?: 'auto' | 'full';
}
```

Defaults are `hover` and `auto` for now.

These props give more flexibility to Next.js' `Link` component. They do deviate from their `Link` api. Their `prefetch` prop IMO is hard to follow (eg: `false` and `undefined` doing different things). Adding `hover` was going to make this prop even weirder, went the explicit way.

### CODE REVIEW

**Comment by bookernath** [comment]
Love this approach!

**Comment by christensenep** [comment]
Played around with this and it seems pretty nice. Delay is still noticable if you're in "click around and see how quickly the site responds" mode, but for a normal shopper I would imagine this is *nearly* as good as the more aggressive prefetching.

I tested and from what I can tell clicking partway through an inflight prefetch does *not* trigger another request, which is what I was hoping I would discover. 👍

File: apps/core/components/link/index.tsx [hunk 1]
```diff
@@ -1,21 +1,56 @@
...
+
+interface PrefetchOptions {
+  prefetch?: 'hover' | 'viewport' | 'none';
+  prefetchKind?: 'auto' | 'full';
+}
+
+type Props = NextLinkProps & PrefetchOptions;
 
+/**
+ * This custom `Link` component extends Next.js's `Link` with additional prefetching controls, making
```
Comment (christensenep) [inline apps/core/components/link/index.tsx:20]: It took me a bit to realize that `auto` and `full` were describing the difference between Next.js's `undefined` and `true` values, I think that could probably be better explained here. That said, this is probably fine for now as I think ultimately we're going to probably want to reference the catalyst.dev/docs here (as this kind of thing will certainly live there), which we should have linkable soon enough, so we can iterate there instead.

**Review by christensenep — APPROVED** [review]

**Review by chanceaclark — APPROVED** [review]

**Review by jorgemoya — APPROVED** [review]
So basically what we expected to happen (prefetch on hover)
